### PR TITLE
Run3-alca218B Clean some of the Hcal calibration code in the test area of Calibration/HcalCalibalgos

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Clean some of the Hcal calibration code in the test area of Calibration/HcalCalibalgos

#### PR validation:

Use test workflows in Calibration/HcalCalibAlgos/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

File: Calibration/HcalCalibAlgos/test/DiJetAnalyzer.cc [hunk 1]
```diff
@@ -45,34 +219,31 @@ void DiJetAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& evS
...
   if (!pfjets.isValid()) {
     throw edm::Exception(edm::errors::ProductNotFound)
         << " could not find PFJetCollection named " << pfJetCollName_ << ".\n";
     return;
   }
 
   // Get RecHits in HB and HE
-  edm::Handle<edm::SortedCollection<HBHERecHit, edm::StrictWeakOrdering<HBHERecHit>>> hbhereco;
-  iEvent.getByToken(tok_HBHE_, hbhereco);
+  edm::Handle<edm::SortedCollection<HBHERecHit, edm::StrictWeakOrdering<HBHERecHit>>> hbhereco =
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/test/DiJetAnalyzer.cc:230]:
```suggestion
  const edm::Handle<edm::SortedCollection<HBHERecHit, edm::StrictWeakOrdering<HBHERecHit>>> hbhereco =
```

File: Calibration/HcalCalibAlgos/test/DiJetAnalyzer.cc [hunk 2]
```diff
@@ -45,34 +219,31 @@ void DiJetAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& evS
...
   if (!hbhereco.isValid()) {
     throw edm::Exception(edm::errors::ProductNotFound)
         << " could not find HBHERecHit named " << hbheRecHitName_ << ".\n";
     return;
   }
 
   // Get RecHits in HF
-  edm::Handle<edm::SortedCollection<HFRecHit, edm::StrictWeakOrdering<HFRecHit>>> hfreco;
-  iEvent.getByToken(tok_HF_, hfreco);
+  edm::Handle<edm::SortedCollection<HFRecHit, edm::StrictWeakOrdering<HFRecHit>>> hfreco = iEvent.getHandle(tok_HF_);
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/test/DiJetAnalyzer.cc:239]:
```suggestion
  const edm::Handle<edm::SortedCollection<HFRecHit, edm::StrictWeakOrdering<HFRecHit>>> hfreco = iEvent.getHandle(tok_HF_);
```

File: Calibration/HcalCalibAlgos/test/GammaJetAnalysis.cc [hunk 3]
```diff
@@ -471,23 +818,15 @@ void GammaJetAnalysis::analyze(const edm::Event& iEvent, const edm::EventSetup&
...
-  iEvent.getByToken(tok_Vertex_, vtxHandle);
-
-  edm::Handle<reco::GsfElectronCollection> gsfElectronHandle;
-  iEvent.getByToken(tok_GsfElec_, gsfElectronHandle);
-  edm::Handle<double> rhoHandle_2012;
-  iEvent.getByToken(tok_Rho_, rhoHandle_2012);
+  edm::Handle<reco::PFCandidateCollection> pfHandle = iEvent.getHandle(tok_PFCand_);
+  edm::Handle<reco::VertexCollection> vtxHandle = iEvent.getHandle(tok_Vertex_);
+  edm::Handle<reco::GsfElectronCollection> gsfElectronHandle = iEvent.getHandle(tok_GsfElec_);
+  edm::Handle<double> rhoHandle_2012 = iEvent.getHandle(tok_Rho_);
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/test/GammaJetAnalysis.cc:824]: all could be `const`

**Comment by tvami** [comment]
@bsunanda I started putting comments about things that can be `const`-s but there is so many I stopped doing it. Please change them to const

**Comment by bsunanda** [comment]
@tvami I used const in as many places as possible. Some collection handles can be kept without const to match the calls to the relevant functions

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by yuanchao** [comment]
+1
* code cleaning for Calibration/HcalCalibAlgos/test/
* WF matrix tests and unit tests all passed
* Maybe take the chance to remove this? (though not part of the PR)
```
Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc:449:9: warning: Value stored to 'tmpsect' is never read [deadcode.DeadStores]
        tmpsect = 1;

```

**Comment by perrotta** [comment]
+1
- (Everything in the test area)

**Pull request merged**